sklearn/decomposition/_nmf.py

```diff
@@ -1272,23 +1221,14 @@ def _check_w_h(self, X, W, H, update_H):
             )
         return W, H
 
-    def _scale_regularization(self, X):
-        """Scale regularization terms."""
+    def _compute_regularization(self, X):
+        """Compute scaled regularization terms."""
         n_samples, n_features = X.shape
-        if self.alpha_W != 0 or self.alpha_H != "same":
-            # if alpha_W or alpha_H is not left to its default value we ignore alpha
-            # and regularization, and we scale the regularization terms.
-            l1_reg_W = n_features * self._l1_reg_W
-            l1_reg_H = n_samples * self._l1_reg_H
-            l2_reg_W = n_features * self._l2_reg_W
-            l2_reg_H = n_samples * self._l2_reg_H
-        else:
-            # Otherwise we keep the old behavior with no scaling
-            # TODO remove in 1.2
-            l1_reg_W = self._l1_reg_W
-            l1_reg_H = self._l1_reg_H
-            l2_reg_W = self._l2_reg_W
-            l2_reg_H = self._l2_reg_H
+
```

Comment:
I prefer not to have `self._alpha_W` attribute if we only need it when computing the regularization. We can define it here and use it immediately:

```python
alpha_W = self.alpha_W
alpha_H = self.alpha_W if self.alpha_H == "same" else self.alpha_H

l1_reg_W = n_features * alpha_W * self.l1_ratio
```

(This would also save on the size of the model when serialized)